### ISSUE TITLE
arcFace的代码错了

### ISSUE THREAD

**Issue by MagicXiaoJing** [MccreeZhao/FR-Loss-on-Mnist#2]
def forward(self, x, label):
        cosine = F.normalize(x).mm(F.normalize(self.weight, dim=0))
        sine = torch.sqrt(1.0 - torch.pow(cosine, 2))
        phi = cosine * self.cos_m - sine * self.sin_m  # 两角和公式
        # 为了保证cos(theta+m)在0-pi单调递减：
        phi = torch.where((cosine - self.th) > 0, phi, cosine - self.mm)
        one_hot = torch.zeros_like(cosine)
        one_hot.scatter_(1, label.view(-1, 1), 1)
        output = (one_hot * phi) + ((1.0 - one_hot) * cosine)
        output = output * self.s
        #这里这个s比较关键，如果设置的不够大会导致收敛困难，
        # 比较容易出现两个类别的Feature重叠的现象，
        # 这也是加性Margin相对于乘性Margin的一个比较弱势的地方，
        # 对于特征向量的收缩要求足够，但是对两类特征向量之间的距离约束不够
        loss = F.cross_entropy(cosine, label)

        return loss, output

在这段 loss = F.cross_entropy(cosine, label)应该写成loss = F.cross_entropy(output, label), 但是如果这样写的话loss一下就成nan了， 我也在解决这个问题， 感觉就像lenet++不适合arcface一样

**Comment by Sunzz1996** [comment]
同样遇到这个问题

**Comment by MagicXiaoJing** [comment]
感觉像是数据集太简单， 轻松拟合了， 然后就开始各种inf，后来相同的代码直接切到res50+ms1m-v1,就不会出现这个问题

**Comment by Sunzz1996** [comment]
直接换复杂的数据吗？但是数据集简单就出现Nan也没道理啊

**Comment by MagicXiaoJing** [comment]
![image](https://user-images.githubusercontent.com/5450325/65829726-d6712d00-e2da-11e9-9d5d-0d61b53ff8b6.png)

**Comment by MagicXiaoJing** [comment]
懒得打字了， 就直接截图了， 以前主要原因是这个。

**Comment by MagicXiaoJing** [comment]
所以还是数据集太简单， 导致input与weight方向几乎完全相同，所以反向的时候就出现了inf

**Comment by Sunzz1996** [comment]
原来是这样子！受教了受教了！感恩~~
